pulsar-function-go/pf/stats_test.go

```diff
@@ -0,0 +1,222 @@
+	"github.com/stretchr/testify/assert"
+)
+
+/*func test(){
+	var metrics_label_names = []string{"tenant", "namespace", "name", "instance_id", "cluster", "fqfn"}
+	var exception_label_names = []string{"error", "ts"}
+	var exception_metrics_label_names = append(metrics_label_names, exception_label_names...)
+	var stat_process_latency_ms = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name: PULSAR_FUNCTION_METRICS_PREFIX + PROCESS_LATENCY_MS,
+			Help: "Process latency in milliseconds."}, metrics_label_names)
+	var reg *prometheus.Registry
+	reg = prometheus.NewRegistry()
+	reg.MustRegister(stat_process_latency_ms)
+	metrics_labels := []string{"test-tenant","test-tenant/test-namespace", "test-name", "1234", "test-cluster",
+		"test-tenant/test-namespace/test-name"}
+	// 1234 is instanceId
+	// ['test-tenant', 'test-tenant/test-namespace', 'test-name',1234,
+    //    'test-cluster', 'test-tenant/test-namespace/test-name']
+	//var _stat_process_latency_ms = stat_process_latency_ms.WithLabelValues(metrics_labels...)
+	//process_latency_ms_count := stat._stat_process_latency_ms._count.get()
+	//process_latency_ms_sum := stat._stat_process_latency_ms._sum.get()
+}
+func  (stat *StatWithLabelValues) getTotalReceived() float32 {
+	gathering, _ := reg.Gather()
+	out := &bytes.Buffer{}
+	for _, mf := range gathering {
+		if _, err := expfmt.MetricFamilyToText(out, mf); err != nil {
+			panic(err)
+		}
+	}
+	fmt.Print(out.String())
+	fmt.Println("----------")
+}
+*/
+func TestExampleSummaryVec(t *testing.T) {
+
+	temps := prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name:       "pond_temperature_celsius",
+			Help:       "The temperature of the frog pond.",
+			Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+		},
+		[]string{"species"},
+	)
+	// Simulate some observations.
+	for i := 0; i < 1000; i++ {
+		temps.WithLabelValues("litoria-caerulea").Observe(30 + math.Floor(120*math.Sin(float64(i)*0.1))/10)
+		temps.WithLabelValues("lithobates-catesbeianus").Observe(32 + math.Floor(100*math.Cos(float64(i)*0.11))/10)
+	}
+
+	// Create a Summary without any observations.
+	temps.WithLabelValues("leiopelma-hochstetteri")
+
+	// Just for demonstration, let's check the state of the summary vector
+	// by registering it with a custom registry and then let it collect the
+	// metrics.
+	reg := prometheus.NewRegistry()
+	reg.MustRegister(temps)
+
+	metricFamilies, err := reg.Gather()
+	if err != nil || len(metricFamilies) != 1 {
+		panic("unexpected behavior of custom test registry")
+	}
+	match := func(vect *io_prometheus_client.MetricFamily) bool {
+		return *vect.Name == "pond_temperature_celsius"
+	}
+	fiteredMetricFamilies := filter(metricFamilies, match)
+
+	if len(fiteredMetricFamilies) > 1 {
+		panic("Too many metric families")
+	}
+	// Then, we need to filter the metrics in the family to one that matches our label.
+
+	fmt.Println(proto.MarshalTextString(metricFamilies[0]))
```

Comment:
Why don't we use assert to compare whether the expected value and the actual value are equal? If my understanding is correct, this test case will succeed at any time, right?

E.g:
```
assert.Equal(t, x, y)
```